File: packages/server/src/schemas/core/gameObject.ts
import { Schema, type } from "@colyseus/schema";

export class GameObject extends Schema {
  @type("string")
  id: string = ""

  @type("number")
  x: number = 0;

  @type("number")
  y: number = 0;

  @type("number")
  z: number = 0;

  @type("number")
  xVelocity: number = 0;

  @type("number")
  yVelocity: number = 0;

  @type("number")
  zVelocity: number = 0;
  
  width: number = 0;
  height: number = 0;
  constructor() {
    super();
  }
}

File: packages/ai/models/game_state/game_object.py
from pydantic import BaseModel, Field

class GameObject(BaseModel):
    id: str = Field(default="")
    x: float = 0
    y: float = 0
    z: float = 0
    xVelocity: float = 0
    yVelocity: float = 0
    zVelocity: float = 0
    width: float = 0
    height: float = 0
